src/chttpd/test/eunit/chttpd_dbs_info_test.erl

```diff
@@ -69,21 +80,128 @@ dbs_info_test_() ->
         }
     }.
 
-should_return_error_for_get_db_info(Url) ->
+
+get_db_info_should_return_db_info(_) ->
+    DbInfo = fabric:get_db_info("db1"),
+    ?_assertEqual(DbInfo, chttpd_util:get_db_info("db1")).
+
+
+get_db_info_should_return_error_when_db_not_exist(_) ->
+    ?_assertEqual({error, database_does_not_exist},
+        chttpd_util:get_db_info("db_not_exist")).
+
+
+get_db_info_should_return_error_when_time_out(_) ->
     ?_test(begin
-        {ok, Code, _, ResultBody} = test_request:get(
-            Url ++ "/_dbs_info?" ++
-                "keys=[\"db1\"]",
-            [?CONTENT_JSON, ?AUTH]
-        ),
+        meck:new(fabric_util, [passthrough]),
+        meck:expect(fabric_util, request_timeout, fun() -> 0 end),
+        ?assertEqual({error, timeout}, chttpd_util:get_db_info("db1")),
+        meck:unload(fabric_util)
+    end).
+
+
+should_return_error_for_put_dbs_info(Url) ->
+    ?_test(begin
+        {ok, Code, _, ResultBody} = test_request:put(Url
+            ++ "_dbs_info", [?CONTENT_JSON, ?AUTH], ""),
         {Body} = jiffy:decode(ResultBody),
-        [
-            ?assertEqual(
-                <<"method_not_allowed">>,
-                couch_util:get_value(<<"error">>, Body)
-            ),
-            ?assertEqual(405, Code)
-        ]
+        ?assertEqual(405, Code),
+        ?assertEqual(<<"method_not_allowed">>,
+            couch_util:get_value(<<"error">>, Body))
+    end).
+
+
+should_return_dbs_info_for_get_dbs_info(Url) ->
+    ?_test(begin
+        {ok, _, _, ResultBody} = test_request:get(Url
+            ++ "_dbs_info", [?CONTENT_JSON, ?AUTH]),
+        BodyJson = jiffy:decode(ResultBody),
+        {Db1Data} = lists:nth(1, BodyJson),
+        {Db2Data} = lists:nth(2, BodyJson),
+        ?assertEqual(2, length(BodyJson)),
+        ?assertEqual(<<"db1">>, couch_util:get_value(<<"key">>, Db1Data)),
+        ?assertEqual(<<"db2">>, couch_util:get_value(<<"key">>, Db2Data))
+    end).
+
+
+should_return_nothing_when_db_not_exist_for_get_dbs_info(Url) ->
+    ?_test(begin
+        meck:new(chttpd_util, [passthrough]),
+        meck:expect(chttpd_util, get_db_info,
+            fun(_) -> {error, database_does_not_exist} end),
+        {ok, Code, _, ResultBody} = test_request:get(Url
+            ++ "_dbs_info", [?CONTENT_JSON, ?AUTH]),
+        BodyJson = jiffy:decode(ResultBody),
+        ?assertEqual(200, Code),
+        ?assertEqual([], BodyJson),
+        meck:unload(chttpd_util)
+    end).
+
+
+should_return_500_time_out_when_time_is_not_enough_for_get_dbs_info(Url) ->
+    ?_test(begin
+        meck:new(fabric_util, [passthrough]),
```

Comment:
It might better to put this in the `foreach` setup function and meck:unload() in teardown. Otherwise if the test fails it might leave a mocked function around and other tests will start failing after that.